src/v/cloud_topics/frontend/frontend.cc

```diff
@@ -96,11 +96,11 @@ static model::record_batch make_placeholder_batch(
         builder.add_raw_kv(std::nullopt, std::nullopt);
     }
 
+    builder.set_max_timestamp(hdr.max_timestamp);
+    builder.set_timestamp(hdr.first_timestamp);
+    builder.set_base_sequence(hdr.base_sequence);
```

Comment:
This will still be wrong for local timequeries, when we create records we have to preserve the timestamp delta, and I don't know what the batch builder does but it won't preserve the input batch correctly. I think we have to rewrite this function to do a custom builder. Or make a new builder that is way more generic...

Reply:
i think we should separate concerns. this fixes the CRC, it sounds like you are identifying a bigger issue with the semantic contents of the batch (now that we are looking at it) ?